platforms/jvm/language-java/src/integTest/groovy/org/gradle/api/tasks/compile/JavaCompileIntegrationTest.groovy

```diff
@@ -317,6 +317,33 @@ class JavaCompileIntegrationTest extends AbstractIntegrationSpec {
         noExceptionThrown()
     }
 
+    def "gradle classpath does not leak onto java compile classpath"() {
```

Comment:
Do we need to enforce that this runs with a non-forking compiler?

I still don't understand why it did not reproduce for me when executed as a forking integration test.

Reply:
By default, the Java compiler runs in process when it can.  We could move this to `JavaCompilerIntegrationSpec` which has subclasses that expect daemon/in-process/forked compilation.  It has checks that make sure we're running in process vs not.